contracts/cw-proposal-single/src/contract.rs

```diff
@@ -413,16 +428,12 @@ pub fn execute_close(
     let old_status = prop.status;
 
     let refund_message = match &prop.deposit_info {
-        Some(deposit_info) => {
-            let receiver = if deposit_info.refund_failed_proposals {
-                &prop.proposer
-            } else {
-                // If we aren't refunding failed proposals then return
-                // the depost to the DAO treasury on close.
-                &config.dao
-            };
-            get_return_deposit_msg(deposit_info, receiver)?
-        }
+        Some(deposit_info) => match deposit_info.refund_policy {
+            // If we are always refunding deposits, funds go to proposer
+            DepositRefundPolicy::Always => get_return_deposit_msg(deposit_info, &prop.proposer)?,
+            // If no refunds, or failed proposals not refund, funds go to DAO
+            _ => get_return_deposit_msg(deposit_info, &config.dao)?,
```

Comment:
These match statements might be nice to actually fully write out. It's probably a desirable feature that adding a new variant will cause a compile error and force us to check this logic again. I do love how clean this is though lol

```suggestion
            // If no refunds, or failed proposals not refund, funds go to DAO
            DepositRefundPolicy::Never | DepositRefundPolicy::OnlyPassed => get_return_deposit_msg(deposit_info, &config.dao)?,
```